File: CPPLib/src/UnixWrapper.cpp
#include "comms/GenericNetWrapper.hpp"
#include "comms/UnixWrapper.hpp"

int newUnixServer(const std::string& path) {
    int server_fd = socket(AF_UNIX, SOCK_STREAM, 0);

    sockaddr_un addr{};
    addr.sun_family = AF_UNIX;
    strncpy(addr.sun_path, path.c_str(), sizeof(addr.sun_path)-1);

    unlink(path.c_str());
    bind(server_fd, (sockaddr*)&addr, sizeof(addr));
    listen(server_fd, 1);

    return server_fd;
}

File: PythonLib/DEFCOM/_UNIXWrapper.py
#!/usr/bin/env python3
import socket
import os
import time
from ._GenericNetWrapper import genericNetWrapper

class clientUnixCon(genericNetWrapper):
  def __init__(self,FILE):
    conn = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
    conn.connect(FILE)

    super().__init__(conn)


def newUnixServer(FILE):
  if os.path.exists(FILE):
    os.remove(FILE)
  s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
  s.bind(FILE)

  s.listen(1)
  return s  
    
    
class serverUnixCon(genericNetWrapper):
      def __init__(self,Server):
        conn, addr = Server.accept()

        super().__init__(conn, addr)
